### ISSUE TITLE
Ancillaries Component `Failed to load extras`

### ISSUE THREAD

**Issue by mattsafaii** [duffelhq/duffel-components#184]
Following my previous issue of getting an "HTMLElement not defined error" when importing `@duffel/components`, I decided to try using the CDN instead. The Ancillaries component is rendering, and works completely fine when I use `fixture_off_1`.

The issue is when I pass in the Offer ID and Client key, I get an error in the UI of the rendered Ancillaries component that says `Failed to load extras`
`An unknown error occurred while retrieving the offer.`

I am using version 3.1.5 and dynamically importing it into my Sveltekit application from the CDN using:
`https://assets.duffel.com/components/3.1.5/duffel-ancillaries.js`

I am also using the multi-step search. When I am on the Fare Selection step, I am setting the selected fare option to the `$selectedOffer` store and passing the Offer ID using `$selectedOffer.id`, and I am passing the Client Key using `$fareOptions.client_key`.

Here is the full code I am using for this component:
```
<script>
  import { onMount } from 'svelte';
  import { selectedOffer, fareOptions } from '$lib/stores';

  onMount(() => {
    // Dynamically create and append the script tag
    const script = document.createElement('script');
    script.src = 'https://assets.duffel.com/components/3.1.5/duffel-ancillaries.js';

    script.onload = () => {
      // Initialize duffel-ancillaries once the script is loaded
      const duffelAncillariesElement = document.querySelector('duffel-ancillaries');
      duffelAncillariesElement.render({
          offer_id: $selectedOffer.id,
          client_key: $fareOptions.client_key,
          services: ['bags', 'seats', 'cancel_for_any_reason'],
          passengers: [
            {
                given_name: "Mae",
                family_name: "Jemison",
                gender: "F",
                title: "dr",
                born_on: "1956-10-17",
                email: "m.jemison@nasa.gov",
                phone_number: "+16177562626",
            }
          ],
      debug: true,
      });

      duffelAncillariesElement.addEventListener('onPayloadReady', (event) => {
        console.log('onPayloadReady\n', event.detail);
      });
    };

    document.head.appendChild(script);
  });
</script>

<!-- Use the custom element -->
<duffel-ancillaries></duffel-ancillaries>
```

The expectation is for the Ancillaries component to work the same way when I use `fixture_off_1` and when I pass in an Offer ID and Client Key.

Here is a screenshot of what the component looks like with the error:
![CleanShot 2023-12-09 at 00 53 17@2x](https://github.com/duffelhq/duffel-components/assets/3858718/9f9be94e-f7ac-4510-a1f8-0886de9d4b67)

Here is a screenshot of what I get in the console from the debug option:
![CleanShot 2023-12-09 at 00 55 05@2x](https://github.com/duffelhq/duffel-components/assets/3858718/d2957571-6acf-4a10-81f4-68eb835eee8b)

And here is the expanded version of the debug log in the console:
![CleanShot 2023-12-09 at 01 07 44@2x](https://github.com/duffelhq/duffel-components/assets/3858718/679734dc-551e-43d5-a87b-d5e2e90eaeb6)

Additional info:
I am in test mode. For the flight options, I selected round-trip, 1 passenger, and economy class. I also selected Duffel Airways for the Outbound and Inbound flights.

As you can see in the screenshot, the Ancillaries component is receiving an offer ID, client key, and passengers object with 1 passenger.

I don't know what could be causing this error and would appreciate any help.

Please let me know if there is any additional info that I should provide to solve this.

Thanks!

**Comment by igorp1** [comment]
Hi @mattsafaii, thank you for your patience with this issue. I was able to make some time to dig into it today and I've noticed that our API does not seem to return the bags and cancel for any reason services as it should. 

I'll raise it with the team and open a PR to fix it on our API.  

I've also opened a [PR](https://github.com/duffelhq/duffel-components/pull/193) to add a little guard to prevent the component from totally crashing. This should be available when on the next release, I'll come back here to keep you posted. 

In the meantime though, to unblock you, I'd recommend fetching the offer on your server and then passing it into the component. I'll bypass this issue. If you also provide the client key along with it, it can fetch the seat maps for you.

**Comment by Dutchgoose** [comment]
Hello @igorp1 I ran into the same issue with the Node.JS library. This workaround was a charm, however in development I all of the ancillaries show "not available".

Is there any way to go through a testing seat map and baggage adding in dev?

**Comment by igorp1** [comment]
> in development I all of the ancillaries show "not available".

My immediate guess is that the flag to request the offer with its services is missing. When you call `GET /air/offers/:id` make sure you include the query param `return_available_services` set to `true`. Seat maps is a bit different. you'd need to fetch that separately as well [`GET https://api.duffel.com/air/seat_maps`](https://duffel.com/docs/api/v1/seat-maps/get-seat-maps). If you work with "Duffel Airways" offers we'll always return a seat map, so maybe you can try that as well. 

I was finally able to make some time to fix the backend issue I had mentioned. so the component should be working with just the client key offer id and passengers. 

@Dutchgoose, @mattsafaii, I'll close this issue up now since I believe the problem is resolved, but if you come across any other problems feel free to open a new issue or re-open this one :)
